src/Microsoft.Framework.PackageManager/Restore/NuGet/NuGetv2Feed.cs

```diff
@@ -232,17 +240,49 @@ public PackageInfo BuildModel(string id, XElement element)
             });
         }
 
+        private void EnsureValidFindPackagesResponse(XDocument doc)
+        {
+            if (!string.Equals(doc.Root.GetDefaultNamespace().ToString(), _defaultNamespace,
+                StringComparison.OrdinalIgnoreCase))
```

Comment:
This is semantically not correct and can return `false` for a valid NuGet xml file since it relies on Xml being serialized the way you expect and when using namespaces there are at least, In most cases when an element is in a namespace it actually does not have the the default namespace since the namespace the element belongs to is either inherited from it's parent or the namespace-uri is bound to a namespace prefix. Therefore you need to get the namespace from the element name and not the default namespace that is declared along with attributes and other (non-default) namespaces.